app/services/jwt_token.rb

```diff
@@ -36,6 +36,9 @@ def to_s
   # This method does not verify if the token signature is valid
   def decoded_payload
     @decoded_payload ||= JWT.decode(token, nil, false).first
+  rescue JWT::DecodeError => e
+    Foreman::Logging.exception('Failed to decode JWT', e)
+    nil
```

Comment:
This nil won't be cached so the exception is logged every time. It's used. Is this something we want to address?

Reply:
I don't think this is called a lot many times. Still I mean, I am not sure what should be the correct solution for this. Open to suggestions.

Reply:
https://jtway.co/how-to-memoize-false-and-nil-values-fe28b8ede9f8